File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.cpp
class Solution {
public:
    int findClosestNumber(vector<int>& nums) {
        int mini=INT_MAX;
        unordered_map<int,int>mp;
        for(auto it:nums){
            int x=abs(it-0);
            mp[it]=x;
            mini=min(mini,x);
        }
        for(auto it:mp){
            if(it.second==mini){
                if(mp.count(abs(mini)))return abs(it.first);
                return it.first;
            }
        }
        return -1;
    }
};

File: 2350-find-closest-number-to-zero/2350-find-closest-number-to-zero.java
public class Solution {
    public int findClosestNumber(int[] nums) {
        int mini = Integer.MAX_VALUE;
        Map<Integer, Integer> mp = new HashMap<>();
        
        for (int num : nums) {
            mp.put(num, Math.abs(num));
            mini = Math.min(mini, Math.abs(num));
        }
        
        for (Map.Entry<Integer, Integer> entry : mp.entrySet()) {
            if (entry.getValue() == mini) {
                if (mp.containsKey(Math.abs(mini))) return Math.abs(entry.getKey());
                return entry.getKey();
            }
        }
        
        return -1;
    }
}
